src/server.c

```diff
@@ -6553,7 +6563,7 @@ void dismissMemory(void* ptr, size_t size_hint) {
 void dismissClientMemory(client *c) {
     /* Dismiss client query buffer and static reply buffer. */
     dismissMemory(c->buf, c->buf_usable_size);
-    dismissSds(c->querybuf);
+    if (c->querybuf) dismissSds(c->querybuf);
```

Comment:
I don't know which pattern is more established but `sdsfree` and `dismissMemory` work with NULL ptrs. @madolson should we align with `sdsfree` instead?

Reply:
I don't know how pervasive it was being adopted, but the preference was that all free functions should accept NULL pointers. So I guess we can be consistent:
```suggestion
    dismissSds(c->querybuf);
```

I know there was mentions in the past that if the compiler doesn't end up inlining this code, it will unnecessary evict memory from the instruction cache since it will load the first instruction + a cache line of some random place in memory.

Reply:
`dismissSds` calls `sdsAllocPtr` which doesn't handle NULL ptrs.
We can modify `dismissSds` to handle NULL. but in this case it would be reasonable to modify all the `dismiss<Type>` function family. Perhaps this should be done as a dedicated PR?

Reply:
If you really want it in a separate PR, I'm okay with that.